lib/dndfonts.sty

```diff
@@ -1,20 +1,68 @@
+\ExplSyntaxOn
+
 \ifpkglayout
   \RequirePackage{bookman}
   \RequirePackage[T1]{fontenc}
   \renewcommand{\sfdefault}{lmss}
 \fi
 
-\newcommand{\dnd@TitleFont}{\normalfont\scshape}
+\keys_define:nn { dnd / fonts }
+  {
+    % Section titles
+    title .tl_set:N         = \l__dnd_font_title_tl,
+    title .initial:n        = \normalfont \scshape,
+    title .value_required:n = true,
+    % Tables
+    table-title .tl_set:N         = \l__dnd_font_table_title_tl,
+    table-title .initial:n        = \fontfamily{fosj} \selectfont \bfseries \scshape,
+    table-title .value_required:n = true,
+    table-body .tl_set:N         = \l__dnd_font_table_body_tl,
+    table-body .initial:n        = \sffamily,
+    table-body .value_required:n = true,
+    % Comment box
+    box-title .tl_set:N         = \l__dnd_font_box_title_tl,
+    box-title .initial:n        = \fontfamily{fosj} \selectfont \bfseries \scshape,
+    box-title .value_required:n = true,
+    box-body .tl_set:N         = \l__dnd_font_box_body_tl,
+    box-body .initial:n        = \sffamily,
+    box-body .value_required:n = true,
+    % Stat blocks
+    stat-block-title .tl_set:N         = \l__dnd_font_stat_block_title_tl,
+    stat-block-title .initial:n        = \normalfont \bfseries \scshape,
+    stat-block-title .value_required:n = true,
+    stat-block-section .tl_set:N         = \l__dnd_font_stat_block_section_tl,
+    stat-block-section .initial:n        = \fontfamily{fosj} \selectfont \scshape,
+    stat-block-section .value_required:n = true,
+    stat-block-body .tl_set:N         = \l__dnd_font_stat_block_body_tl,
+    stat-block-body .initial:n        = \sffamily,
+    stat-block-body .value_required:n = true,
+    % Footer
+    footer .tl_set:N         = \l__dnd_font_footer_tl,
+    footer .initial:n        = \normalfont \scshape,
+    footer .value_required:n = true,
+    page-number .tl_set:N         = \l__dnd_font_page_number_tl,
+    page-number .initial:n        = \normalfont,
+    page-number .value_required:n = true,
+  }
+
+\NewDocumentCommand { \DndSetFonts } { o }
+  {
+    \keys_set:nn { dnd / fonts } { o }
+  }
+
+% \newcommand{\dnd@TitleFont}{\normalfont\scshape}
```

Comment:
I think the best way of dealing with the change in the font system is to cause a breaking change here. We cannot use a deprecate warning because the user will have used `\renewcommand` to replace the font definition, removing the deprecate warning.